x-pack/filebeat/input/netflow/netflow_test.go

```diff
@@ -60,6 +67,108 @@ type TestResult struct {
 	Flows []beat.Event `json:"events,omitempty"`
 }
 
+func newV2Context() (v2.Context, func()) {
+	ctx, cancel := context.WithCancel(context.Background())
+	return v2.Context{
+		Logger:      logp.NewLogger("netflow_test"),
+		ID:          "test_id",
+		Cancelation: ctx,
+	}, cancel
+}
+
+func TestNetFlow(t *testing.T) {
+	pcaps, err := filepath.Glob(filepath.Join(pcapDir, "*.pcap"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, file := range pcaps {
+		testName := strings.TrimSuffix(filepath.Base(file), ".pcap")
+
+		t.Run(testName, func(t *testing.T) {
+
+			pluginCfg, err := conf.NewConfigFrom(mapstr.M{})
+			require.NoError(t, err)
+
+			netflowPlugin, err := Plugin(logp.NewLogger("netflow_test")).Manager.Create(pluginCfg)
+			require.NoError(t, err)
+
+			mockPipeline := &pipelinemock.MockPipelineConnector{}
+
+			ctx, cancelFn := newV2Context()
+			errChan := make(chan error)
+			go func() {
+				defer close(errChan)
+				errChan <- netflowPlugin.Run(ctx, mockPipeline)
+			}()
+
+			defer cancelFn()
+
+			require.Eventually(t, mockPipeline.HasConnectedClients, 5*time.Second, 100*time.Millisecond,
+				"client connects to pipeline")
+
+			udpAddr, err := net.ResolveUDPAddr("udp", defaultConfig.Config.Host)
+			require.NoError(t, err)
+
+			conn, err := net.DialUDP("udp", nil, udpAddr)
+			require.NoError(t, err)
+
+			f, err := pcap.OpenOffline(file)
+			require.NoError(t, err)
+			defer f.Close()
+
+			goldenData := readGoldenFile(t, filepath.Join(goldenDir, testName+".pcap.golden.json"))
+
+			// Process packets in PCAP and get flow records.
+			var totalBytes, totalPackets int
+			packetSource := gopacket.NewPacketSource(f, f.LinkType())
+			for pkt := range packetSource.Packets() {
+				payloadData := pkt.TransportLayer().LayerPayload()
+
+				n, err := conn.Write(payloadData)
+				require.NoError(t, err)
+				totalBytes += n
+				totalPackets++
+			}
+
+			require.Eventually(t, func() bool {
+				return len(mockPipeline.GetAllEvents()) == len(goldenData.Flows)
+			}, 5*time.Second, 100*time.Millisecond,
+				"see all expected events in pipeline")
```

Comment:
Same thing here.
```suggestion
			require.Eventually(t, func() bool {
				return len(mockPipeline.GetAllEvents()) == len(goldenData.Flows)
			}, 5*time.Second, 100*time.Millisecond,
				"got a different number of events than expected")
```

Reply:
@belimawr please re-review after the above changes